flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesMultipleComponentLeaderElectionDriver.java

```diff
@@ -0,0 +1,268 @@
+                    new KubernetesException(
+                            String.format(
+                                    "ConfigMap %s does not exist. This indicates that somebody has interfered with Flink's operation.",
+                                    configMapName)));
+            return false;
+        }
+    }
+
+    @Override
+    public void publishLeaderInformation(String componentId, LeaderInformation leaderInformation)
+            throws Exception {
+        Preconditions.checkState(running.get());
+
+        kubeClient
+                .checkAndUpdateConfigMap(
+                        configMapName,
+                        updateConfigMapWithLeaderInformation(componentId, leaderInformation))
+                .get();
+
+        LOG.debug(
+                "Successfully wrote leader information {} for leader {} into the config map {}.",
+                leaderInformation,
+                componentId,
+                configMapName);
+    }
+
+    @Override
+    public void deleteLeaderInformation(String componentId) throws Exception {
+        publishLeaderInformation(componentId, LeaderInformation.empty());
+    }
+
+    private Function<KubernetesConfigMap, Optional<KubernetesConfigMap>>
+            updateConfigMapWithLeaderInformation(
+                    String leaderName, LeaderInformation leaderInformation) {
+        final String configMapDataKey = KubernetesUtils.createSingleLeaderKey(leaderName);
+
+        return kubernetesConfigMap -> {
+            if (KubernetesLeaderElector.hasLeadership(kubernetesConfigMap, lockIdentity)) {
+                final Map<String, String> data = kubernetesConfigMap.getData();
+
+                if (leaderInformation.isEmpty()) {
+                    data.remove(configMapDataKey);
+                } else {
+                    data.put(
+                            configMapDataKey,
+                            KubernetesUtils.encodeLeaderInformation(leaderInformation));
+                }
+
+                kubernetesConfigMap.getLabels().putAll(configMapLabels);
+                return Optional.of(kubernetesConfigMap);
+            }
+
+            return Optional.empty();
+        };
+    }
+
+    private static Collection<LeaderInformationWithComponentId> extractLeaderInformation(
+            KubernetesConfigMap configMap) {
+        final Map<String, String> data = configMap.getData();
+
+        final Collection<LeaderInformationWithComponentId> leaderInformationWithLeaderNames =
+                new ArrayList<>();
+
+        for (Map.Entry<String, String> keyValuePair : data.entrySet()) {
+            final String key = keyValuePair.getKey();
+            if (KubernetesUtils.isSingleLeaderKey(key)) {
+                final String leaderName = KubernetesUtils.extractLeaderName(key);
+                final LeaderInformation leaderInformation =
+                        KubernetesUtils.parseLeaderInformationSafely(keyValuePair.getValue());
```

Comment:
I think this is ok because the corresponding `DefaultLeaderElectionService` would issue another publish command if it is the leader and its leader information is empty.